packages/repository/src/relations/has-many/has-many-through.helpers.ts

```diff
@@ -109,6 +117,45 @@ export function createThroughConstraint<Through extends Entity, ForeignKeyType>(
   const constraint: any = {[sourceFkName]: fkValue};
   return constraint;
 }
+/**
+ * Creates constraint used to create the through model
+ *
+ * @param relationMeta - resolved hasManyThrough metadata
+ * @param targetInstance instance of target entity used to constrain through
+ * @internal
+ *
+ * @example
+ * ```ts
+ * const resolvedMetadata = {
+ *  // .. other props
+ *  keyFrom: 'id',
+ *  keyTo: 'id',
+ *  through: {
+ *    model: () => CategoryProductLink,
+ *    keyFrom: 'categoryId',
+ *    keyTo: 'productId',
+ *  },
+ * };
+ * createThroughConstraint(resolvedMetadata, {id: 3, name: 'a product'});
+ *
+ * >>> {'productId': 1}
+ *
+ *  * createThroughConstraint(resolvedMetadata, {id: {inq:[3,4]}});
+ *
+ * >>> {'productId': {inq:[3,4]}}
+ */
+export function createThroughFkConstraint<Target, Through extends Entity>(
```

Comment:
This new helper is created for deleting through models once the target instances get deleted